internals_fsutils.go

```diff
@@ -382,53 +386,51 @@ func createZip(archiveName string, files map[string][]byte) error {
 		if err != nil {
 			return err
 		}
-		_, err = f.Write([]byte(fcont))
+		_, err = io.Copy(f, fcont)
 		if err != nil {
 			return err
 		}
 	}
 
-	// Make sure to check the error on Close.
-	err := w.Close()
-	if err != nil {
-		return err
-	}
-
-	err = ioutil.WriteFile(archiveName, buf.Bytes(), defaultFilePermissions)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	// The Close call writes the checksum, so return the error
```

Comment:
Glad it was helpful! See [named result parameters](https://golang.org/doc/effective_go.html#named-results) for an explanation. They work well with `defer`, because the deferred function is allowed to change return values.